Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -598,33 +600,33 @@ namespace AZ
             }
             else
             {
-                if (!imageAsset.IsReady())
+                AZ::Data::AssetInfo assetInfo;
+                AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, imageAsset.GetId());
+
+                Data::Instance<Image> image = nullptr;
+                if (assetInfo.m_assetType == azrtti_typeid<StreamingImageAsset>())
                 {
-                    imageAsset = Data::AssetManager::Instance().GetAsset<StreamingImageAsset>(imageAsset.GetId(), AZ::Data::AssetLoadBehavior::PreLoad);
-                    imageAsset.BlockUntilLoadComplete();
-                    if (!imageAsset.IsReady())
-                    {
-                        AZ_Error(s_debugTraceName, false, "Image asset could not be loaded");
-                        return false;
-                    }
+                    Data::Asset<StreamingImageAsset> streamingImageAsset{imageAsset.GetId(), assetInfo.m_assetType};
```

Comment:
IIRC making a new asset reference like this can cause issues, though I don't remember what the problem was exactly. the Asset<> class has a GetAs function that you should consider using here:
```suggestion
                    StreamingImageAsset* streamingImageAsset = imageAsset.GetAs<StreamingImageAsset>();
```

Reply:
I can't use the second one since it's not a DataAsset<> which we can't used it for FindOrCreate()

I think the issue you thought was that if the imageAsset had a un-matching assetType, it would report an error when Asset::SetAssetData

Reply:
I updated to 
`Data::Asset<StreamingImageAsset> streamingImageAsset = Data::static_pointer_cast<StreamingImageAsset>(imageAsset);`